### PR TITLE
New sprites for gases

### PR DESCRIPTION
**Pull request by nicbn**
[Changelogs]: #
:cl:
imageadd: Gases changed.
/:cl:

https://www.youtube.com/watch?v=pVTZxBr78Yg (N2, O2, air and CO2 are still invisible, I replaced the gases in the chambers)
New N2O: https://www.youtube.com/watch?v=IJdHU1DO6kg

Also gas opacity changes depending on the amount of gas that there is

### CODE REVIEW

**Comment by epochayur** [comment]
god damn some of these are good, A+

**Comment by Vidiveni** [comment]
Will carbon dioxide still be invisible in high enough densities to knock people out?

File: code/__DEFINES/atmospherics.dm [hunk 1]
```diff
@@ -246,6 +249,13 @@
 
 #define ASSERT_GAS(gas_id, gas_mixture) if (!gas_mixture.gases[gas_id]) { ADD_GAS(gas_id, gas_mixture.gases) };
 
+//prefer this to gas_mixture/total_moles in performance critical areas
+#define TOTAL_MOLES(cached_gases, out_var)\
+	out_var = 0;\
```
Comment (ninjanomnom) [inline code/__DEFINES/atmospherics.dm]: Didn't we remove this define before because it wasn't worth having around

Comment (Iamgoofball) [reply code/__DEFINES/atmospherics.dm]: but its getting used here

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 2]
```diff
@@ -113,17 +113,21 @@
...
-			var/gas_meta = gas[GAS_META]
-			var/gas_overlay = gas_meta[META_GAS_OVERLAY]
-			if(gas_overlay && gas[MOLES] > gas_meta[META_GAS_MOLES_VISIBLE])
-				. += gas_overlay
+	if(!air)
+		return
+	. = new /list
+	var/list/gases = air.gases
+	var/total_moles = 0
+	TOTAL_MOLES(gases, total_moles)
```
Comment (ninjanomnom) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm]: You don't seem to be using `total_moles` at all

**Review by ninjanomnom — DISMISSED** [review]

File: code/__DEFINES/atmospherics.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/__DEFINES/atmospherics.dm]: I was misremembering removing some other define for atmos we removed due to maintainability concerns even though it was marginally faster.

**Comment by subject217** [comment]
Will kill the glitter scare once people get used to it. Still quite nice looking.

**Comment by Barhandar** [comment]
This is frankly ugly and against the point of oxygen/nitrogen invisibility.

**Comment by MrDoomBringer** [comment]
@Barhandar it should reduce the visibility of things u dip

**Comment by evsey9** [comment]
How do multiple gasses on one tile look?

Also, are we gonna get oxygen/nitrogen sprites?

**Comment by nicbn** [comment]
CO2 is always invisible.

And nitrogen and oxygen are also obviously invisible, as they should be (I can't tell if you are being sarcastic).

Multiple gases will blend one over another like they already do.

I kept the glitter with the old sprites, but if you want a way to do the scare thing you can add it later - it's out of the scope of this PR.

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm (hunk 2 above)
Comment (nicbn) [reply code/modules/atmospherics/environmental/LINDA_turf_tile.dm]: Will fix in a moment

**Comment by subject217** [comment]
@barhandar he put other gases in the o2/n2/airmix/etc tanks, all this does is resprite existing gas overlays

**Comment by nicbn** [comment]
Oh sorry I didn't understand. Yeah I didn't change N2, O2, CO2 or air sprites, I changed the gases in the chambers. They're miasma, nitryl, water vapour and hyper-noblium.

**Comment by Vidiveni** [comment]
During a round (with the test merge) where many gases mixed and was put on fire, the tiles turned completely white. I think it was plasma, water, pluoxium and some other stuff. Happened on the meteor escape shuttle near end of round. Lexia Black was throwing plasma smoke omega-weed.

**Comment by nicbn** [comment]
Tritium fire is very white. The gases may have blended with it, making it even more opaque and whiter. This is kind of expected and shouldn't be a big problem as will only happen if someone purposefully floods a tile with multiple gases.

File: code/__DEFINES/atmospherics.dm [hunk 3]
```diff
@@ -62,7 +62,10 @@
 //GASES
 #define MIN_TOXIC_GAS_DAMAGE				1
 #define MAX_TOXIC_GAS_DAMAGE				10
-#define MOLES_GAS_VISIBLE					0.5		//Moles in a standard cell after which gases are visible
+#define MOLES_GAS_VISIBLE					0.25	//Moles in a standard cell after which gases are visible
```
Comment (duncathan) [inline code/__DEFINES/atmospherics.dm]: please revert this and other whitespace changes in the file

**Review by duncathan — CHANGES_REQUESTED** [review]
looks good otherwise and very pretty

File: code/__DEFINES/atmospherics.dm (hunk 3 above)
Comment (nicbn) [reply code/__DEFINES/atmospherics.dm]: Gases below 0.5 have a really low alpha (13 out of 255). By increasing `MOLES_GAS_VISIBLE`, either the gas "clouds" borders won't blend nicely (as there will be no gases with really low alpha) or gases in not-so-high amounts will be too translucent (as the really low alpha will apply to higher amounts of gas).

Comment (MrDoomBringer) [reply code/__DEFINES/atmospherics.dm]: @nicbn he's not talking about the number change, he's talking about the change in number of spaces there are on line 65

Comment (duncathan) [reply code/__DEFINES/atmospherics.dm]: yeah sorry perhaps I was unclear I was referring only to the whitespace change on this line (and the ones above it)

Comment (duncathan) [reply code/__DEFINES/atmospherics.dm]: it's fine tho the other ones were more important

**Review by duncathan — APPROVED** [review]

**Pull request merged**